### ISSUE TITLE
compileComponents() is asynchronous, we must use async()

### ISSUE THREAD

**Issue by wKoza** [angular/angular-cli#4774]
compileComponents () is asynchronous. So, we must use async () method like this:

```javascript
BeforeEach (async (() => {
     TestBed.configureTestingModule ({
       Declarations: [BannerComponent], // declare the test component
     })
     .compileComponents (); // compile template and css
   }));
```
not

```javascript
BeforeEach (() => {
     TestBed.configureTestingModule ({
       Declarations: [BannerComponent], // declare the test component
     })
    TestBed.compileComponents (); // compile template and css
   });
```

So,[ this file](https://github.com/angular/angular-cli/blob/master/packages/%40angular/cli/blueprints/ng2/files/__path__/app/app.component.spec.ts#L6) must be fixed.

**Comment by PascalPrecht** [comment]
While it's semantically true, I think we can get rid off `compileComponents()` entirely since webpack inlines all templates etc. under the hood anyways right?

**Comment by wKoza** [comment]
I m agree. But this pr #4757 has been closed ..

**Comment by clydin** [comment]
Webpack is an internal implementation detail.  The tests shouldn't assume it is being used or that it inlines anything.

**Comment by PascalPrecht** [comment]
@clydin I very much agree. As of right now however, the entire codebase that is generated by angular cli kind of assumes webpack. That's why we don't use `module.id` in component meta data and also relative paths for asses like template and CSS. 

That's why I think, tests can be treated the same way. They are generated by Angular CLI which knows its build process internals.

Just thinking out loud here.

**Comment by clydin** [comment]
You actually can use module.id,  internally it is just unused.  And the relative paths and inlining don't necessitate webpack.  Either can be achieved with other build tool setups.

Either way and in general, I don't think an existing deficiency should be justification for additional.

**Comment by cexbrayat** [comment]
This can be closed now that #4775 has been merged (even if I do agree with @PascalPrecht that getting rid of `compileComponents()` as it's not necessary with the current setup could be a good option).

**Comment by wKoza** [comment]
@cexbrayat , you can re-open  #4757 ;)
